### PR TITLE
[DOC] Add deterministic and probabilistic atlases to glossary

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR proposes to add definitions to glossary for 'probabilistic' and 'deterministic' atlases (definitions might need some improvements/rewording...).
It also updates the docstrings of atlas fetchers to make it more clear what kind of atlases are  returned.

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -47,6 +47,15 @@ If you wish to add a missing term, please `create a new issue`_ or
         `Decoding`_ consists in predicting, from brain images, the conditions
         associated to trial.
 
+    Deterministic atlas
+        A deterministic atlas is a hard parcellation of the brain into
+        non-overlaping regions, which were obtained by a wide range of methods.
```
Comment (bthirion) [inline doc/glossary.rst:52]:
```suggestion
        non-overlaping regions, that might have been obtained by segmentation or clustering methods.
```

File: nilearn/datasets/atlas.py [hunk 2]
```diff
@@ -896,10 +905,11 @@ def fetch_atlas_smith_2009(data_dir=None, mirror='origin', url=None,
 def fetch_atlas_yeo_2011(data_dir=None, url=None, resume=True, verbose=1):
     """Download and return file names for the Yeo 2011 parcellation.
 
-    The provided images are in MNI152 space and have shapes equal to
-    ``(256, 256, 256, 1)``. They contain consecutive integers values
-    from 0 (background) to either 7 or 17 depending on the atlas version
-    considered.
+    This function retrieves
```
Comment (bthirion) [inline nilearn/datasets/atlas.py:908]:
```suggestion
    This function retrieves the so-called yeo
```

**Review by bthirion — COMMENTED** [review]
LGTM, merely some improvements for clarification.

File: nilearn/datasets/atlas.py [hunk 3]
```diff
@@ -146,6 +147,7 @@ def fetch_atlas_difumo(dimension=64, resolution_mm=2, data_dir=None,
 def fetch_atlas_craddock_2012(data_dir=None, url=None, resume=True, verbose=1):
     """Download and return file names for the Craddock 2012 parcellation.
 
+    This function returns :term:`probabilistic atlases<Probabilistic atlas>`.
```
Comment (bthirion) [inline nilearn/datasets/atlas.py:150]:
```suggestion
    This function returns a :term:`probabilistic atlases<Probabilistic atlas>`.
```

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by NicolasGensollen** [comment]
Thanks! Merging...

**Pull request merged**